src/SFA.DAS.EmployerIncentives.Functions.DomainMessageHandlers/ServiceCollectionExtensions.cs

```diff
@@ -84,5 +58,50 @@ public static IServiceCollection AddCommandService(this IServiceCollection servi
 
             return serviceCollection;
         }
+
+        public static IServiceCollection AddNServiceBus(
+            this IServiceCollection serviceCollection,
+            IConfiguration configuration)
+        {
+            var webBuilder = serviceCollection.AddWebJobs(x => { });
+            webBuilder.AddExecutionContextBinding();
+
+            var endpointConfiguration = new EndpointConfiguration("SFA.DAS.EmployerIncentives.Functions.DomainMessageHandlers")
+                .UseMessageConventions()
+                .UseNewtonsoftJsonSerializer()
+                .UseOutbox(true)
+                .UseSqlServerPersistence(() => new SqlConnection(configuration["ApplicationSettings:DbConnectionString"]))
+                .UseUnitOfWork();
+
+            endpointConfiguration.UseContainer<ServicesBuilder>(c => c.ExistingServices(serviceCollection));
+
+            if (configuration["ApplicationSettings:NServiceBusConnectionString"].Equals("UseLearningEndpoint=true", StringComparison.CurrentCultureIgnoreCase))
+            {
+                endpointConfiguration
+                    .UseTransport<LearningTransport>()
+                    .StorageDirectory(configuration.GetValue("ApplicationSettings:UseLearningEndpointStorageDirectory",
+                        Path.Combine(
+                            Directory.GetCurrentDirectory().Substring(0, Directory.GetCurrentDirectory().IndexOf("src")),
+                            @"src\SFA.DAS.EmployerIncentives.Functions.TestConsole\.learningtransport")));
+                endpointConfiguration.UseLearningTransport(s => s.AddRouting());
```

Comment:
TestConsole ?

Reply:
Yes if "UseLearningEndpoint=true" i.e. a test app is calling.